src/components/MyRecipes.vue

```diff
@@ -1,291 +1,566 @@
-                ["clean"] // remove formatting button
-            ]
+
+    async searchIngredientImages(index) {
+      const ingredient = this.ingredients[index].name;
+      const unit = this.ingredients[index].unit;
+      // Remove the existing imageResults property
+      this.$delete(this.ingredients[index], 'imageResults');
+      this.$delete(this.ingredients[index], 'showImageResults');
+      // Check if both name and unit are filled before making the API call
+      if (ingredient && unit) {
+        try {
+          this.$set(this.ingredients[index], 'loading', true); // Set loading state to true
+          // Make the API call to fetch the ingredient thumbnails
+          const response = await this.fetch_ingredient_thumbnail(ingredient)
+          // Extract the thumbnail URLs from the API response
+          const imageResults = response;
+          // Update the corresponding ingredient object with the image results
+          this.$set(this.ingredients[index], 'imageResults', imageResults);
+          this.$set(this.ingredients[index], 'showImageResults', imageResults.length > 0);
+        } catch (error) {
+          console.log(error)
+        } finally {
+          this.$set(this.ingredients[index], 'loading', false);
         }
+      }
     },
-    components: {
-        UploadImage,
-        VueEditor,
-        FlagPicker
+
+    fetch_ingredient_thumbnail(ingredient) {
+      const client = createClient(process.env.PEXEL_API_KEY);
+      const query = ingredient;
+      return new Promise((resolve, reject) => {
+        client.photos.search({ query, per_page: 3 })
+          .then(photos => {
+            resolve(photos.photos)
+          })
+          .catch(error => {
+            console.log(error)
+            reject(error)
+          })
+      })
+    },
+
+    selectThumbnail(index, thumbnailUrl) {
+      this.ingredients[index].thumbnail = thumbnailUrl;
     },
-    methods: {
-        toggleEditor(action) {
-            if (action === 'hide') {
-                $("#recipe-editor").removeClass("show")
-                $("#recipe-editor").addClass("hide")
-            }
-
-            if (action === 'show') {
-                $("#recipe-editor").addClass("show")
-                $("#recipe-editor").removeClass("hide")
-            }
-            this.inEditMode = !this.inEditMode
-        },
-        addField(value, field) {
-            field.push({ value: { name: "", unit: "", thumbnail: "", link: "" } })
-        },
-        removeField(index, field) {
-            field.splice(index, 1);
+
+    dropdownValidation: function () {
+      let isValid = true;
+      if (!this.nationality) {
+        this.nationalityError = "The nationality field is required"
+        isValid = false;
+      }
+      if (!this.cookbook_id) {
+        this.cookbookError = "The cookbook field is required";
+        isValid = false
+      }
+      return isValid
+    },
+
+    async submitButton() {
```

Comment:
I strongly do not see the point of making this call async, what is your reasoning for this? Do you think it necessary? Perhaps I'm missing something? Do you see a problem with making this call async vs sync?